### PR TITLE
Reload functionality clean-up part 1

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

**Comment by wind57** [comment]
started to get my feet wet in reload functionality also, by doing some minor refactoring and changes...

File: spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigurationChangeDetector.java [hunk 1]
```diff
@@ -52,44 +52,26 @@ public abstract class ConfigurationChangeDetector {
...
+		this.strategy = Objects.requireNonNull(strategy);
 	}
 
 	public void reloadProperties() {
-		this.log.info("Reloading using strategy: " + this.strategy.getName());
+		log.info("Reloading using strategy: " + this.strategy.getName());
 		this.strategy.reload();
 	}
 
-	/**
```
Comment (wind57) [inline spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigurationChangeDetector.java:65]: not deleted, but moved a bit down in the class, it's now package private

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigurationChangeDetectorTest.java [hunk 2]
```diff
@@ -1,134 +0,0 @@
-/*
```
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigurationChangeDetectorTest.java:1]: Not deleted, but moved to commons, where it should really belong

File: spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigurationChangeDetector.java [hunk 3]
```diff
@@ -103,41 +84,22 @@ public boolean changed(List<? extends MapPropertySource> left, List<? extends Ma
...
-	 * Finds one registered property source of the given type, logging a warning if
-	 * multiple property sources of that type are available.
-	 * @param <S> property source type
-	 * @param sourceClass class for which property sources will be searched for
-	 * @return matched property source
-	 */
-	protected <S extends PropertySource<?>> S findPropertySource(Class<S> sourceClass) {
```
Comment (wind57) [inline spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigurationChangeDetector.java:113]: no one was using this one

File: spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigurationChangeDetector.java [hunk 4]
```diff
@@ -103,41 +84,22 @@ public boolean changed(List<? extends MapPropertySource> left, List<? extends Ma
...
 	/**
 	 * @param <S> property source type
 	 * @param sourceClass class for which property sources will be found
 	 * @return finds all registered property sources of the given type
 	 */
 	public <S extends PropertySource<?>> List<S> findPropertySources(Class<S> sourceClass) {
-		List<S> managedSources = new LinkedList<>();
+		List<S> managedSources = new ArrayList<>();
 
-		LinkedList<PropertySource<?>> sources = toLinkedList(this.environment.getPropertySources());
```
Comment (wind57) [inline spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigurationChangeDetector.java:132]: I did not understand the purpose of a LinkedList here, when an ArrayList does preserve the same order, as such small refactor here

**Review by ryanjbaxter — APPROVED** [review]

**Pull request merged**